azurelinuxagent/ga/update.py

```diff
@@ -119,6 +147,18 @@ def __init__(self):
 
         self.last_incarnation = None
 
+        self._extensions_summary = ExtensionsSummary()
+
+        self._is_initial_goal_state = not os.path.exists(self._initial_goal_state_file_path())
+
+        if not conf.get_extensions_enabled():
```

Comment:
This initialization used to happen just before the main loop, and the period was a local variable. Now the period is an instance member and is initialized here (plus it takes into account the InitialGoalStatePeriod)